pkg/tfbridge/detailed_diff.go

```diff
@@ -41,6 +41,105 @@ func sortedMergedKeys[K cmp.Ordered, V any, M ~map[K]V](a, b M) []K {
 	return keysSlice
 }
 
+func isTypeShapeMismatched(val resource.PropertyValue, propType shim.ValueType) bool {
+	if !isPresent(val) {
+		return false
+	}
+	switch propType {
```

Comment:
I believe IsOutput implies IsComputed, so both should not be necessary.

Reply:
FWIW Computed and Output are distinct cases.

```
// IsComputed returns true if the underlying value is a computed value.
func (v PropertyValue) IsComputed() bool {
	_, is := v.V.(Computed)
	return is
}

// IsOutput returns true if the underlying value is an output value.
func (v PropertyValue) IsOutput() bool {
	_, is := v.V.(Output)
	return is
}

```

Reply:
``` contract.Assertf(!val.IsComputed() && !val.IsSecret(), "val should not be computed or secret") ```

Asserts are good to have but I don't know why we're not hitting them. Sounds like first-class outputs or secrets for example are bound to show up sometime in the code this is diffing? Why do we think the asserts will not hit? 

What about first-class secrets? If these do not show up it's for some very complicated historical reasons around secrets handling. 

Would it be cheaper just to handle these forms and get to 100% rather than keep chasing this down later on? 

The case analysis is as such. 

1. A value in news may be an unknown. This may be represented as Computed or Output that is unknown

2. A value in either olds or news may be a Secret or an Output with secret=true. For the purposes of this we can drop the secret bit-ness and continue computing differing paths as before, I think. Might worth thinking through this one.

3. Finally a value may be an Output that is secret=false, known=true. Unwrap and proceed on the inner value.